src/main/java/edu/unc/lib/boxc/migration/cdm/services/FindingAidReportService.java

```diff
@@ -201,34 +202,30 @@ private Integer countRecords(String query) {
     }
 
     /**
-     * List of potential collection id values
+     * Map of potential collection id values and the fields they come from
      */
-    private List<String> listPotentialCollectionIds() {
+    private Map<String, String> listPotentialCollectionIds() {
         fieldService.validateFieldsFile(project);
         CdmFieldInfo fieldInfo = fieldService.loadFieldsFromProject(project);
         List<String> exportFields = fieldInfo.listAllExportFields();
 
-        List<String> potentialCollectionIds = new ArrayList<>();
+        Map<String, String> potentialCollectionIds = new HashMap<>();
         // only collect the first 10 potential ids
-        int i = 0;
         try (Connection conn = indexService.openDbConnection()) {
             var stmt = conn.createStatement();
             for (String field : exportFields) {
-                ResultSet rs = stmt.executeQuery(" select '" + field + "' from " + CdmIndexService.TB_NAME
-                        + " where '" + field + "' like " + "'^[A-Za-z0-9]{5}-?z?$'");
+                ResultSet rs = stmt.executeQuery(" select " + field + " from " + CdmIndexService.TB_NAME
+                        + " where " + field + " like " + "'^[A-Za-z0-9]{5}-?z?$' limit 10");
                 while (rs.next()) {
-                    if (!rs.getString(1).isBlank() && i < 10) {
-                        potentialCollectionIds.add(rs.getString(1));
-                        i++;
-                    }
+                    potentialCollectionIds.put(field, rs.getString(1));
                 }
             }
         } catch (SQLException e) {
             throw new MigrationException("Error interacting with export index", e);
         }
 
         if (potentialCollectionIds.isEmpty()) {
-            potentialCollectionIds.add("no potential collection ids found");
```

Comment:
Since there's going to be an `if` either way, maybe it'd make sense to just pass back an empty potentialCollectionIds and then in `collectionReport` have an `if` that checks if the result is empty and displays this message rather than calling `showFieldMapValues`? That would help separate the concerns of generating the list of matches from the presentation of results.